solr/core/src/java/org/apache/solr/api/AnnotatedApi.java

```diff
@@ -335,6 +337,33 @@ void invoke(SolrQueryRequest req, SolrQueryResponse rsp, CommandOperation cmd) {
       }
     }
 
+    public int hashCode() {
```

Comment:
don't you think we should cash and reuse the hash ?

Reply:
Several of the instance variables we use to compute the hashCode here are non-final.  I'd bet those vars are unlikely to change in practice, but I wasn't confident enough in that to put in caching here, just in case.